std/unicode.zig

```diff
@@ -18,11 +18,25 @@ pub fn utf8CodepointSequenceLength(c: u32) !u3 {
 /// returns a number 1-4 indicating the total length of the codepoint in bytes.
 /// If this byte does not match the form of a UTF-8 start byte, returns Utf8InvalidStartByte.
 pub fn utf8ByteSequenceLength(first_byte: u8) !u3 {
-    if (first_byte < 0b10000000) return u3(1);
-    if (first_byte & 0b11100000 == 0b11000000) return u3(2);
-    if (first_byte & 0b11110000 == 0b11100000) return u3(3);
-    if (first_byte & 0b11111000 == 0b11110000) return u3(4);
-    return error.Utf8InvalidStartByte;
+    var ret: u4 = undefined;
+    ret = utf8ByteSequenceLengthNoValidate(first_byte);
+    if (ret > 4)
+        return error.Utf8InvalidStartByte;
+    return @intCast(u3, ret);
+}
+
+pub fn utf8ByteSequenceLengthNoValidate(first_byte: u8) u4 {
+    const utf8_skip_data = []u8{
```

Comment:
Can you show evidence this is faster than the bit manipulations from before? It's quite possible the memory lookups are slower than the bit manipulation instructions. and when there is no perf evidence, the clearer code is the winner, and I think it's fair to say the bit manipulation code here is clearer.

In general it's reasonable to have `utf8ByteSequenceLength` return a possible error, and then tokenizer.zig can use `catch unreachable` if it can guarantee there will be no error.

Reply:
I developed a nice branchless version with `@clz` that I think you will like